autoload/vista/parser/lsp.vim

```diff
@@ -41,80 +43,103 @@ function! s:IsFileUri(uri) abort
   return stridx(a:uri, 'file:///') == 0
 endfunction
 
-" The kind field in the result is a number instead of a readable text, we
-" should transform the number to the symbol text first.
-function! vista#parser#lsp#KindToSymbol(line, container) abort
-  let line = a:line
-  " SymbolInformation interface
-  if has_key(line, 'location')
-    let location = line.location
-    if s:IsFileUri(location.uri)
-      let lnum = location.range.start.line + 1
-      let col = location.range.start.character + 1
-      call add(a:container, {
-         \ 'lnum': lnum,
-         \ 'col': col,
-         \ 'kind': s:Kind2Symbol(line.kind),
-         \ 'text': line.name,
-         \ })
-    endif
-  " DocumentSymbol class
-  elseif has_key(line, 'range')
-    let range = line.range
-    let lnum = range.start.line + 1
-    let col = range.start.character + 1
-    call add(a:container, {
-          \ 'lnum': lnum,
-          \ 'col': col,
-          \ 'kind': s:Kind2Symbol(line.kind),
-          \ 'text': line.name,
-          \ })
-    if has_key(line, 'children')
-      for child in line.children
-        call vista#parser#lsp#KindToSymbol(child, a:container)
-      endfor
+function! s:LspToLocalSymbol(sym, range)
+  return {
+    \ 'lnum': a:range.start.line + 1,
+    \ 'col': a:range.start.character + 1,
+    \ 'kind': s:Kind2Symbol(a:sym.kind),
+    \ 'text': a:sym.name,
+    \ }
+endfunction
+
+function! s:LocalToRawSymbol(sym)
+  return {
+    \ 'line': a:sym.lnum,
+    \ 'kind': a:sym.kind,
+    \ 'name': a:sym.text,
+    \ }
+endfunction
+
+function! s:IsDocumentSymbol(sym)
+  return has_key(a:sym, 'selectionRange')
```

Comment:
Why this change? The author of previous patch about this part uses the `range`.

Reply:
Both `range` and `selectionRange` are mandatory in documentSymbol response, the latter is used for consistency with CoC behavior.